src/muse.c

```diff
@@ -1829,6 +1831,17 @@ find_misc(struct monst* mtmp)
                         }
                     }
     }
+    if (OBJ_AT(mtmp->mx, mtmp->my) && !rn2(2)) {
+        for (otmp = g.level.objects[mtmp->mx][mtmp->my]; otmp; otmp = otmp2) {
+            otmp2 = otmp->nexthere;
+            if (Is_container(otmp) && otmp->otyp != BAG_OF_TRICKS
+                && Has_contents(otmp) && !otmp->olocked && !otmp->otrapped) {
+                g.m.has_misc = MUSE_BAG;
+                g.m.misc = otmp;
+                return TRUE;
```

Comment:
Why does a container being on the monster's current spot override any of the other misc items it can use half of the time?  It feels to me like maybe the `return TRUE` shouldn't be here, so that the other rules for things that the monster actually has a need to use can take precedence further down, and the container looting will only happen as a "well, nothing else interesting to do" thing.  That's basically how bags in inventory work, right?  They are the lowest precedence action in `find_misc`.